uvtools/plot.py

```diff
@@ -466,3 +469,265 @@ def omni_view_gif(filenames, name='omni_movie.gif'):
+
+        # extract parameters
+        ykey, xkey = item[0].split("_vs_") # keys for choosing parameters
+        xvals, yvals = plot_params[xkey], plot_params[ykey]
+
+        # get labels
+        xlabel, ylabel = labels[xkey], labels[ykey]
+
+        # plot stuff
+        for ax, diff in zip(axes[i], diffs):
+            # set labels
+            ax.set_xlabel(xlabel, fontsize=12)
+            ax.set_ylabel(ylabel, fontsize=12)
+
+            # plot waterfall and add a colorbar
+            fig.sca(ax)
+            cax = waterfall(diff, mode="real", cmap='viridis', 
+                            extent=extent(xvals, yvals))
+            fig.colorbar(cax)
+
+    # display the figure
+    plt.tight_layout()
+    plt.show()
+
+    # save if desired
+    if save_path is not None:
+        utils.savefig(fig, save_path)
+
+def plot_diff_uv(uvd1, uvd2, pol=None, speedup=True,
+                 skip_check=False, save_path=None,
+                 resolution=50):
+    """Summary plot for difference between visibilities.
+
+    Parameters
+    ----------
+    uvd1, uvd2 : pyuvdata.UVData
+        Input UVData objects which contain the visibilities to be differenced 
+        and any relevant metadata.
+
+    pol : str, None, optional
+        String specifying which polarization to be used. Must be one of the
+        polarizations listed in the UVData.get_pols() method for both
+        `uvd1` and `uvd2`. Default is to use the 0th polarization.
+
+    speedup : bool, optional
+        Whether to use the fast implementation of this plotting tool or the
+        slower implementation. The faster implementation does not produce
+        figures as nice as the slower implementation, but the slower
+        implementation is only really usable for small sets of data.
+        Default is to use the fast implementation.
+
+    skip_check : bool, optional
+        Whether to check that the metadata for `uvd1` and `uvd2` match.
+        Default behavior is to check the metadata.
+
+    save_path : str, optional
+        Path to where the figure should be saved; may be absolute or relative.
+        Default is to not save the figure.
+    
+    resolution : int, optional
+        Number of bins to use for regridding the u and v arrays.
+
+    """
+    # check the metadata unless instructed otherwise
+    if not skip_check:
+        utils.check_uvd_pair_metadata(uvd1, uvd2)
+
+    # convert polarization to index
+    pol = 0 if pol is None else uvd1.get_pols().index(pol)
+
+    # load in relevant metadata
+    bl_vecs = uvd1.uvw_array
+    freqs = uvd1.freq_array[0]
+
+    # import astropy constants to convert freq to wavelength
+    from astropy.constants import c
+    wavelengths = c.value / freqs
+
+    # get uvw vectors; shape = (Nfreq, Nblts, 3)
+    uvw_vecs = np.array([bl_vecs / wavelength for wavelength in wavelengths])
+    
+    # reshape uvw vectors to (Nblts, Nfreq, 3)
+    uvw_vecs = np.einsum("ijk->jik", uvw_vecs)
```

Comment:
`einsum` is occasionally stupid and does operations in a brute force way. Might be worth checking if `numpy.swapaxes` is faster in this particular case.